lib/kubeclient/config.rb

```diff
@@ -0,0 +1,111 @@
+require 'yaml'
+require 'base64'
+
+module Kubeclient
+  class Config
+    class Context
+      attr_reader :api_endpoint, :api_version, :ssl_options
+
+      def initialize(api_endpoint, api_version, ssl_options)
+        @api_endpoint = api_endpoint
+        @api_version = api_version
+        @ssl_options = ssl_options
+      end
+    end
```

Comment:
For convenience, it would be nice to have a `#client` method of context that would return a `Kubeclient::Client`, so you could say:

```ruby
config = Kubeclient::Config.read(ENV['HOME'] + '/.kube/config')
client = config.context('vagrant').client
```